tests/integration/remote_configuration_tester/src/charm.py

```diff
@@ -0,0 +1,68 @@
+#!/usr/bin/env python3
+# Copyright 2022 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""A Charm to functionally test the Alertmanager Operator."""
+
+import logging
+
+import yaml
+from charms.alertmanager_k8s.v0.alertmanager_remote_configuration import (
+    ConfigReadError,
+    RemoteConfigurationProvider,
+    load_config_file,
+)
+from ops.charm import CharmBase, PebbleReadyEvent
+from ops.main import main
+from ops.model import ActiveStatus, WaitingStatus
+
+logger = logging.getLogger(__name__)
+
+
+class AlertmanagerTesterCharm(CharmBase):
+    """A Charm to functionally test the Alertmanager Operator."""
+
+    ALERTMANAGER_CONFIG_FILE = "/etc/alertmanager/alertmanager.yml"
```

Comment:
It seems to me that with every Juju event temp files are removed. This means I'd have to create this temp file and write config to it in the constructor of the charm. Wouldn't that be ugly comparing to having a storage?

Reply:
50/50. I mean, yes, the tempfile would be removed. But it also doesn't need to be `tempfile.TemporaryFile()`. 

It could also just be a dict dumped to YAML, right where this is, and on `pebble_ready`, push it into the container, just like you're doing now. Pebble will happily push [just a `str](https://github.com/canonical/operator/blob/main/ops/model.py#L1548)